tests/foreman/api/test_puppet_classes.py

```diff
@@ -0,0 +1,102 @@
+"""Unit tests for the ``Puppet Classes`` paths.
+
+A full API reference for PuppetClasses can be found here:
+http://theforeman.org/api/apidoc/v2/puppetclasses.html
+
+"""
+from ddt import ddt
+from fauxfactory import gen_string
+from nailgun import client, entities
+from random import randint
+from robottelo.common.decorators import data, run_only_on
+from robottelo.common.helpers import get_server_credentials
+from robottelo.test import APITestCase
+
+
+@ddt
+class PuppetClassesTestCase(APITestCase):
+    """Tests for ``katello/api/v2/puppetclasses``."""
+    @run_only_on('sat')
+    @data(
+        {u'name': gen_string('alphanumeric', randint(1, 255))},
+        {u'name': gen_string('alpha', randint(1, 255))},
+        {u'name': gen_string('cjk', randint(1, 85))},
+        {u'name': gen_string('latin1', randint(1, 255))},
+        {u'name': gen_string('numeric', randint(1, 255))},
+        {u'name': gen_string('utf8', randint(1, 85))}
+    )
+    def test_positive_create_puppet_class_1(self, attrs):
+        """@Test: Create a puppet-class and provide a name.
+
+        @Assert: A puppet-class is created with the provided attributes.
+
+        @Feature: Puppet Classes
+
+        """
+        puppet_class_id = entities.PuppetClass(**attrs).create_json()['id']
```

Comment:
Yes.... the read and create works and returns entities.puppetclass object but for verification i used create_json() which will get me the puppet_class id and read_json() which will return the dictionary of puppet class items.